x-pack/plugins/observability_solution/apm/public/components/shared/entity_enablement/index.tsx

```diff
@@ -77,17 +76,17 @@ export function EntityEnablement({ label, tooltip }: { label: string; tooltip?:
         }
         refetch();
       } else {
-        if (response.reason === ERROR_USER_NOT_AUTHORIZED) {
-          setIsLoading(false);
-          setsIsUnauthorizedModalVisible(true);
-          return;
-        }
-
         throw new Error(response.message);
       }
     } catch (error) {
       setIsLoading(false);
       const err = error as Error | IHttpFetchError<ResponseErrorBody>;
+      if ('response' in err) {
+        if (err.response?.status === 403) {
+          setsIsUnauthorizedModalVisible(true);
+          return;
+        }
+      }
```

Comment:
@elastic/obs-ux-infra_services-team I wanted to do the same change for the disablement logic but didn't find any reference for `disableManagedEntityDiscovery()`, we don't plan on using it yet ?

Reply:
I wonder if we shouldn't throw a custom error for this, maybe `EntityManagerUnauthorizedError`. Then they could just use `if (error instanceof EntityManagerUnauthorizedError)` and it would look cleaner. We could export the `EntityManagerUnauthorizedError` from the `kbn-entities-schema` package.

Reply:
>I wonder if we shouldn't throw a custom error for this, maybe EntityManagerUnauthorizedError. Then they could just use if (error instanceof EntityManagerUnauthorizedError) and it would look cleaner

That would be better indeed.

Reply:
>I wanted to do the same change for the disablement logic but didn't find any reference for disableManagedEntityDiscovery()

I'm not aware of this logic @klacabane , can you elaborate, please?

Reply:
@cauemarcondes there is a way to enable EEM but also disable it (stop and uninstall transforms and all associated components). I was expecting the UI to call the disable endpoint when the user returns to the classic view

Reply:
@klacabane that was the initial implementation but we changed that. The bug [here ](https://github.com/elastic/observability-dev/issues/3736)  (internal)

Reply:
I wonder if we should give users a way to stop and uninstall transforms as well. Since it was created automatically, users might think that switching back to classic would delete it, which is not true.

cc: @roshan-elastic

Reply:
Hey all,

(cc @chrisdistasio) - we decided to not build an explicit mechanism for 'disabling' EEM because the intention is to turn it on for everyone by default.

Having said that, we have provided some docs to explain what it is and how to turn it off using the Dev Tools:

https://www.elastic.co/docs/current/serverless/observability/new-experience-services
https://www.elastic.co/docs/current/serverless/observability/elastic-entity-model

A question on the scope of this issue (i.e. '[eem] return 403 if user is not authorized to enable entity discovery'), is there a time when this would surface?

My understanding is that we have an explicit workflow for when someone doesn't have permission to enable EEM build into the new experience:

- [ ] https://github.com/elastic/kibana/issues/187568

Reply:
> A question on the scope of this issue (i.e. '[eem] return 403 if user is not authorized to enable entity discovery'), is there a time when this would surface?

this change has no functional impact on the current workflow, it's just a nicer way to handle the unauthorized scenario